services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java

```diff
@@ -104,7 +106,14 @@ public News createNews(News news, org.exoplatform.services.security.Identity cur
       }
       News createdNews;
       if (PublicationDefaultStates.PUBLISHED.equals(news.getPublicationState())) {
-        createdNews = postNews(news, currentIdentity.getUserId());
+        try {
+          createdNews = newsStorage.getNewsById(news.getId(), false);
+        } catch (ItemNotFoundException e) {
+          createdNews = newsStorage.createNews(news);
+        }
+        if (createdNews != null) {
+          createdNews = postNews(news, currentIdentity.getUserId());
+        }
       } else if (news.getSchedulePostDate() != null){
         createdNews = unScheduleNews(news, currentIdentity);
```

Comment:
@hakermi normally with the same scenario we will also a server exception when we try to schedule a deleted draft article, so we should fix that also

Reply:
Ok